Improve handling of CSS syntax errors

tuskpot opened the pull request:
With this change, se tools print a more descriptive error message when cssselect encounters a CSS syntax error during lint or build.

Previously, the scripts simply exited with a stack trace that was difficult to decipher, and didn't indicate where in the CSS the problem was. This was reported in [this email](https://groups.google.com/d/msg/standardebooks/0CW04wKdKKE/S-gXUGW0DAAJ) on the mailing list.

This update allows se to exit with a clear error message that specifically says that there is a CSS problem and prints the text of the line where the problem was encountered, like this:
> Error: Couldn't parse CSS in or near this line: p[epub|type~"z3998:salutation"]

Limitations of this approach:
* The error doesn't say which file the error was in. Since SE projects should only be editing CSS in local.css, this is probably fine.
* Some kinds of CSS errors are detected on the previous or next line, which can cause unhelpful messages like "Couldn't parse CSS in or near this line: }". I don't know how to improve this. (Maybe print the lines before and after the error? I'm not sure how to do that nicely with the se.print_error function.)
* The original exception from cssselect actually says which character the parser was on when it identified the error. It would be cool to point this out in the error message, but not cool enough for the extra work it would involve.

acabal commented:
Great idea, thanks! 

The build function should not print to the console, it should instead throw exceptions that the caller can catch and print if necessary. So, can you tweak this so that exceptions are raised instead? You can see a list of SE exceptions in `__init__.py`, or you can add a new SE exception if none of those look right. Check the rest of the build function for examples of how to throw SE exceptions with error messages.

tuskpot commented:
Ah, of course. Thank you.

I've added a new se.InvalidCssException, and the build script now raises that instead of printing to console.

I also modified the error message to print the original exception message, similar to what the build script does when it raises se.InvalidXhtmlException. This gives a little extra context, which is sometimes very helpful:

> Error: Couldn't parse CSS in or near this line: p[epub|type~"z3998:salutation"]
> Operator expected, got <DELIM '~' at 11>

acabal commented:
lint should also raise an exception. I see that it currently does print & exit at one point, this is an error and I'll fix that.

tuskpot commented:
Got it. Lint now raises the same exception.

acabal commented:
Awesome, thank you!
